src/infrastructure/content/ContentCommunicator.java

```diff
@@ -1,24 +1,143 @@
+	 * @return mapImage - The HashMap which stores the userName and userImage
+	 */
+	protected HashMap<String, String> getMapImage() {
+		return mapImage;
+	}
+	
+	/**
+	 * This method allows other classes within this package to set the HashMap mapImage
+	 * @param mapImage - The HashMap which stores the userName and userImage of all clients
+	 */
+	protected void setMapImage(HashMap<String, String> mapImage) {
+		this.mapImage = mapImage;
+	}
+	
+	/**
+	 * This method provides HashMap mapHandler to other classes within this package
+	 * @return mapHandler - The HashMap which stores the identifier and handler of subscriber of this module
+	 */
+	protected HashMap<String, IContentNotificationHandler> getMapHandler() {
+		return mapHandler;
+	}
+	
+	/**
+	 * This method initializes a user by storing its username and userimage in a map as Strings
+	 * This method stores these values locally too and generates server's ipAddress by getting
+	 * 		the port value of the board from ServerPort class
+	 * Finally, some manipulation on JSON string is done and then sent to server via networking
+	 * @param userDetails - This is a JSON string which contains a JSON object with three fields, 
+	 * 		namely ipAddress, username and image.
+	 */
 	public void initialiseUser(String userDetails) {
-    	
-    }
+		JSONObject jsonObject = new JSONObject(userDetails);
+		serverIPAddress = jsonObject.getString("ipAddress");
+		userName = jsonObject.getString("username");
+		userImage = jsonObject.getString("image");
+		ServerPort serverPort = new ServerPort();
+		port = serverPort.getPort();
+		serverIPAddress += ":";
+		serverIPAddress += String.valueOf(port);
+		mapImage.put(userName, userImage);
+		jsonObject.put("meta", "newUser");
+		jsonObject.remove("ipAddress");
+		communicator.send(serverIPAddress, jsonObject.toString(), "contentServer");
+	}
 	
-	@Override
+	/**
+	 * This method gets the message from UI module and then send it via networking module
+	 * A meta field is added to it indicating that it is message before sending to server
+	 * @param message - This is a JSON string which contains a JSON object with key "message"
+	 */
 	public void sendMessageToContent(String message) {
-    	
-    }
+		JSONObject jsonObject = new JSONObject(message);
+		jsonObject.put("meta", "message");
+		communicator.send(serverIPAddress, jsonObject.toString(), "contentServer");
+	}
 	
-	@Override
+	/**
+	 * This method clears all the maps created by content module locally and informs the
+	 * server that the client is leaving the board by sending a JSON String which contains
+	 * a JSON object with two keys, meta and username via networking module
+	 */
 	public void notifyUserExit() {
-    	
-    }
-    
-	@Override
+		mapImage.clear();
+		mapHandler.clear();
+		JSONObject jsonObject = new JSONObject();
+		jsonObject.put("meta", "userExit");
+		jsonObject.put("username", userName);
+		communicator.send(serverIPAddress, jsonObject.toString(), "contentServer");
+	}

```

Comment:
Same problem here as given above. The three functions (messagetoContent, notifyUserexit and the next one) are UI-content interfaces. Please check this.

Reply:
What is the problem if it is within the methods of UI-Content interface methods?

Reply:
ContentCommunicator is purely my implementation and I can do anything which is required. Where does this restriction arrive to not include networking methods in UI-Content interface methods?